python/cudf/cudf/testing/_utils.py

```diff
@@ -16,6 +16,8 @@
 from cudf.core.column.datetime import _numpy_to_pandas_conversion
 from cudf.utils import dtypes as dtypeutils
 
+_JOIN_TYPES = ("left", "inner", "outer", "right", "leftanti", "leftsemi")
```

Comment:
Feels a bit weird that we're defining join-specific utilities here. From what I can tell, `assert_join_results_equal` is used in `test_joning.py`, but also a few tests in `test_string.py` use it. 

Can those tests simply be moved to `test_joining.py`? That way we can keep these utilities in `test_joining.py`.

Reply:
Yup, moved this Ashwin.